apps/testnorge-arena/arena-core/src/main/java/no/nav/registre/arena/core/consumer/rs/command/PostRettighetCommand.java

```diff
@@ -0,0 +1,52 @@
+package no.nav.registre.arena.core.consumer.rs.command;
+
+import lombok.extern.slf4j.Slf4j;
+import no.nav.registre.arena.core.consumer.rs.request.RettighetRequest;
+import no.nav.registre.testnorge.domain.dto.arena.testnorge.vedtak.NyttVedtakResponse;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+import java.util.concurrent.Callable;
+
+import static no.nav.registre.arena.core.consumer.rs.util.Headers.CALL_ID;
+import static no.nav.registre.arena.core.consumer.rs.util.Headers.CONSUMER_ID;
+import static no.nav.registre.arena.core.consumer.rs.util.Headers.NAV_CALL_ID;
+import static no.nav.registre.arena.core.consumer.rs.util.Headers.NAV_CONSUMER_ID;
+
+@Slf4j
+public class PostRettighetCommand implements Callable<NyttVedtakResponse> {
+
+    private final WebClient webClient;
+    private final RettighetRequest rettighet;
+
+    public PostRettighetCommand(RettighetRequest rettighet, WebClient webClient) {
+        this.webClient = webClient;
+        this.rettighet = rettighet;
+    }
+
+    @Override
+    public NyttVedtakResponse call() {
+        NyttVedtakResponse response = null;
```

Comment:
Hvorfor ikke bare retunere exception hvis det blir feil?

Eller rekoverer du et annet sted i koden siden du retunerer null?

Reply:
Ja, det blir håndtert i consumeren. Det er lettere å ikke returnere exception siden det kan være en lang liste med rettigheter som sendes inn for flere forskjellige identer. Hvis en rettighet feiler så kan da resten komme gjennom.